libs/payments/eligibility/src/lib/eligibility.manager.spec.ts

```diff
@@ -262,4 +273,223 @@ describe('EligibilityManager', () => {
+                  items: [
+                    EligibilityContentSubgroupOfferingResultFactory({
+                      stripeProductId: 'prod_test',
+                    }),
+                    EligibilityContentSubgroupOfferingResultFactory({
+                      stripeProductId: 'prod_test2',
+                    }),
+                  ],
+                },
+              }),
+            ],
+          },
+        },
+      });
+      mockOfferingResult.getOffering = jest
+        .fn()
+        .mockReturnValueOnce(mockTargetOfferingResult);
+      const result = await manager.getProductIdOverlap(
+        ['prod_test2'],
+        mockTargetOfferingResult
+      );
+      expect(result.length).toBe(1);
+      expect(result[0].comparison).toBe(OfferingComparison.DOWNGRADE);
+    });
+
+    it('should return same comparison for same productId', async () => {
+      const mockTargetOfferingResult = EligibilityContentOfferingResultFactory({
+        stripeProductId: 'prod_test',
+      });
+      const existingResult = EligibilityContentOfferingResultFactory({
+        stripeProductId: 'prod_test',
+      });
+      mockOfferingResult.getOffering = jest
+        .fn()
+        .mockReturnValueOnce(mockTargetOfferingResult)
+        .mockReturnValueOnce(existingResult);
+      const result = await manager.getProductIdOverlap(
+        ['prod_test'],
+        mockTargetOfferingResult
+      );
+      expect(result.length).toBe(1);
+      expect(result[0].comparison).toBe(OfferingComparison.SAME);
+    });
+
+    it('should return upgrade comparison for upgrade planId', async () => {
+      const mockTargetOfferingResult = EligibilityContentOfferingResultFactory({
+        stripeProductId: 'prod_test2',
+        linkedFrom: {
+          subGroupCollection: {
+            items: [
+              EligibilityContentSubgroupResultFactory({
+                offeringCollection: {
+                  items: [
+                    EligibilityContentSubgroupOfferingResultFactory({
+                      stripeProductId: 'prod_test',
+                    }),
+                    EligibilityContentSubgroupOfferingResultFactory({
+                      stripeProductId: 'prod_test2',
+                    }),
+                  ],
+                },
+              }),
+            ],
+          },
+        },
+      });
+      const existingResult = EligibilityContentOfferingResultFactory({
+        stripeProductId: 'prod_test',
+      });
+      mockOfferingResult.getOffering = jest
+        .fn()
+        .mockReturnValueOnce(mockTargetOfferingResult)
+        .mockReturnValueOnce(existingResult);
+      const result = await manager.getProductIdOverlap(
+        ['prod_test'],
+        mockTargetOfferingResult
+      );
+      expect(result.length).toBe(1);
+      expect(result[0].comparison).toBe(OfferingComparison.UPGRADE);
+    });
+
+    // it('should return multiple comparisons in multiple subgroups', async () => {
```

Comment:
nit: should this use [`it.skip()`](https://jestjs.io/docs/api#testskipname-fn) so we remember to reenable the tests?

Reply:
Peterrrrrrrr, I'm still working on this pull request lol